config/vars.go

```diff
@@ -0,0 +1,81 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/undefinedlabs/go-env"
+)
+
+var (
+	current *ScopeConfig
+	m       sync.RWMutex
+)
+
+func Get() *ScopeConfig {
+	// We check is already loaded with a reader lock
+	m.RLock()
+	if current != nil {
+		defer m.RUnlock()
+		return current
+	}
+	m.RUnlock()
+
+	// Is not loaded we block to load it
+	m.Lock()
```

Comment:
I'd have this initialization explicitely in the agent initialization, rather than in the Get method.

Reply:
The configuration could be used before the agent initialization, the `autoinstrument` import needs to know if some patches are gonna be used, and that occurs before initializing the agent.

That's why I prefer the lazy loading on demand.